src/query/compiler.js

```diff
@@ -736,7 +722,29 @@ assign(QueryCompiler.prototype, {
   },
 
   // "Preps" the update.
-  _prepUpdate(data) {
+  _prepUpdate(data = {}) {
+    const { counter = [] } = this.single;
+    const grouped = groupBy(counter, 'column');
+
+    for (const column in grouped) {
+      //Skip?
+      if (has(data, column)) {
+        //Needed?
+        this.client.logger.warn(
```

Comment:
Most warns have to do with feature support by different SQL engines;
whereas errors are generally thrown for incorrect query construction. I'd say throw an error.
(just glancing at source/tests)

Reply:
I think its bold to assume calling `.increments` first and then `.update` with the same column to be a user mistake, it may just as likely be perfectly intentional.

No strong opinion on what should happen here, but definitely some form of message must be spit out of knex instead of silently ignoring. I opted for warn over error due to the argument above, but not 100% against changing this.